SemantIQ RTD Provider: initial release

alexandr-kim-vl opened the pull request:
## Type of change
- [x] Feature

- [x] Does this change affect user-facing APIs or examples documented on http://prebid.org?
  https://github.com/prebid/prebid.github.io/pull/5904

## Description of change

Adding a new RTD submodule that receives targeting data from the SemantIQ service and populates `ortb2Fragments`.

Maintainer: [alexandr.kim@audienzz.com](mailto:alexandr.kim@audienzz.com) 

### Module configuration
```
  pbjs.setConfig({
    ...
    realTimeData: {
      dataProviders: [
        {
          name: 'semantiq',
          waitForIt: true,
          params: {
            companyId: 12345,
            timeout: 1000,
          },
        },
      ],
    },
  });
```

ChrisHuie commented:
@alexandr-kim-vl we have fixed the build error in the current version of master if you bring in recent commits

osazos commented:
Hi, 

In your doc, you specify that the `companyId` is required. However, there is no mechanism to ensure that it has been set, and providing feedback on this could be beneficial.

To optimize the byte usage for each request, consider the following suggestions:
- Currently, you set a `none` string when there is no value for a keyword. Would it be acceptable to simply omit the keyword altogether, or does `none` hold specific significance for your customers?
- You duplicate the keyword for each value, e.g., `ctx_segment=C001,ctx_segment=C002`. Could it be an option to consolidate these into a single keyword with concatenated values, such as `ctx_segment=C001|C002`? I am unsure of the actual usage, so this suggestion might not be applicable.

Let me know, otherwise LGTM.

alexandr-kim-vl commented:
Hi @osazos!

Thank you for the review and recommendations.

> In your doc, you specify that the `companyId` is required. However, there is no mechanism to ensure that it has been set, and providing feedback on this could be beneficial.

You're right, marked it as optional.

> Currently, you set a `none` string when there is no value for a keyword. Would it be acceptable to simply omit the keyword altogether, or does `none` hold specific significance for your customers?

I removed the keywords with no value and adjusted the tests, it shouldn't make any difference.

> You duplicate the keyword for each value, e.g., `ctx_segment=C001,ctx_segment=C002`. Could it be an option to consolidate these into a single keyword with concatenated values, such as `ctx_segment=C001|C002`? I am unsure of the actual usage, so this suggestion might not be applicable.

This format is chosen because it's the only way to pass an array of values to the appnexusBidAdapter, and Xandr is our main partner right now. But I'll keep an eye on what format the other bidders use and may come up with a more optimal option.

osazos commented:
@alexandr-kim-vl, did you plan to add a doc in https://github.com/prebid/prebid.github.io? You should add it asap to be able to merged.

alexandr-kim-vl commented:
> @alexandr-kim-vl, did you plan to add a doc in https://github.com/prebid/prebid.github.io? You should add it asap to be able to merged.

Wanted to wait for feedback before creating it: https://github.com/prebid/prebid.github.io/pull/5904

osazos commented:
> @osazos wanted to wait for feedback before creating it: [prebid/prebid.github.io#5904](https://github.com/prebid/prebid.github.io/pull/5904)

Add a comment for that in the doc PR plz.

alexandr-kim-vl commented:
> > @osazos wanted to wait for feedback before creating it: [prebid/prebid.github.io#5904](https://github.com/prebid/prebid.github.io/pull/5904)
> 
> Add a comment for that in the doc PR plz.

@osazos I meant feedback on the module PR, the documentation is now ready, thanks again for checking.
